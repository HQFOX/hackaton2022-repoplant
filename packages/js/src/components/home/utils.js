import {
  Level1,
  Level2Average,
  Level3Bad,
  Level4,
  Level5
} from "@hv/uikit-react-icons";

export const getStatus = statusNumber => {
  switch (statusNumber) {
    case 1:
      return { Icon: Level1, sema: "sema10" };
    case 2:
      return { Icon: Level2Average, sema: "sema11" };
    case 3:
      return { Icon: Level3Bad, sema: "sema12" };
    case 4:
      return { Icon: Level4, sema: "sema13" };
    case 5:
      return { Icon: Level5, sema: "sema14" };
    default:
      return { Icon: null, sema: "sema1" };
  }
};
