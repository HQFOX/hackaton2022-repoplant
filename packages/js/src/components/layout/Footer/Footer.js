import React from "react";
import HvFooter from "@hv/uikit-react-lab/dist/Footer";

const Footer = () => <HvFooter />;

export default Footer;
