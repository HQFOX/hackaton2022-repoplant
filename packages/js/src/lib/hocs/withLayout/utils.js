export const findById = (data = [], id) => {
  for (let i = 0; i < data.length; i += 1) {
    const el = data[i];
    if (el.id === id) return el;

    const child = findById(el.data, id);
    if (child) return child;
  }
  return null;
};

export const findLeaf = item => (item.data ? findLeaf(item.data[0]) : item);

export const findFirstChildByParentId = (data = [], id) => {
  const parent = findById(data, id);
  return (parent && parent.data && findLeaf(parent.data[0]).id) || id;
};

export const navigationDepth = arr =>
  Array.isArray(arr)
    ? 1 + Math.max(...arr.map(el => navigationDepth(el.data)))
    : 0;
