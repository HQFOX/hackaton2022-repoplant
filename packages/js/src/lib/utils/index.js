export const getRandom = (max, min = 1) =>
  Math.floor(Math.random() * (max - min + 1)) + min;
