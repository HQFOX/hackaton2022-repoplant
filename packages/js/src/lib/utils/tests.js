import React from "react";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import configureMockStore from "redux-mock-store";
import { createMemoryHistory } from "history";
import HvProvider from "@hv/uikit-react-core/dist/Provider";

const mockStore = configureMockStore([thunk]);
const memoryHistory = createMemoryHistory();

const createMockStore = mockState => mockStore(mockState);

const withStoreProvider = (Component, store) => props => (
  <Provider store={createMockStore(store)}>
    <HvProvider>
      <Component {...props} />
    </HvProvider>
  </Provider>
);

const withHvProvider = Component => props => (
  <HvProvider>
    <Component {...props} />
  </HvProvider>
);

export {
  createMockStore,
  mockStore,
  memoryHistory,
  withStoreProvider,
  withHvProvider
};
