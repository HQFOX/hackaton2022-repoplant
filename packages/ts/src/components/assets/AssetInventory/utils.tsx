import React from "react";
import {
  AssetInventoryMetadata,
  AssetInventoryConfiguration,
  HvSemanticColorKeys
} from "@hv/uikit-react-core";
import {
  Delete,
  Preview,
  Upload,
  Level1,
  Level2Average,
  Level3Bad,
  Level4,
  Level5
} from "@hv/uikit-react-icons";

const metadata: AssetInventoryMetadata[] = [
  {
    id: "id1",
    accessor: "headerTitle",
    cellType: "alpha-numeric",
    searchable: true,
    sortable: true,
    sortableLabelAsc: "Title ascending",
    sortableLabelDesc: "Title descending"
  },
  {
    id: "id2",
    accessor: "semantic",
    cellType: "alpha-numeric"
  },
  {
    id: "id3",
    accessor: "probability",
    cellType: "numeric",
    searchable: true,
    sortable: true,
    sortableLabelAsc: "Probability ascending",
    sortableLabelDesc: "Probability descending"
  },
  {
    id: "id4",
    accessor: "timeHorizon",
    cellType: "numeric",
    sortable: true,
    sortableLabelAsc: "TimeHorizon ascending",
    sortableLabelDesc: "TimeHorizon descending"
  },
  {
    id: "id5",
    accessor: "event.schedule",
    cellType: "alpha-numeric",
    searchable: true
  },
  {
    id: "id6",
    accessor: "event.description",
    cellType: "alpha-numeric",
    searchable: true
  }
];

const viewConfiguration: AssetInventoryConfiguration = {
  breakpoints: { xs: 12, sm: 6, md: 4, lg: 3, xl: 3 },
  columnConfiguration: [
    { style: { width: 1, textAlign: "center" } },
    {
      title: "Event",
      style: { minWidth: "360px", textAlign: "start" }
    },
    {
      title: "Probability",
      style: { minWidth: "80px", textAlign: "end" }
    },
    {
      title: "Time horizon",
      style: { minWidth: "100px", textAlign: "end" }
    },
    {
      title: "Related Assets",
      style: {
        minWidth: "195px",
        paddingLeft: "30px",
        textAlign: "start"
      }
    }
  ]
};

const configuration = {
  metadata,
  viewConfiguration
};

const actions = [
  { id: "post", label: "Dismiss", disabled: false },
  {
    id: "get",
    label: "Preview",
    icon: <Preview color="atmo5" />,
    disabled: true
  },
  {
    id: "put",
    label: "Upload",
    icon: <Upload color="atmo5" />,
    disabled: true
  },
  { id: "delete", label: "Delete", icon: <Delete />, disabled: false }
];

type Status = {
  Icon: React.ComponentType<any> | null;
  sema: HvSemanticColorKeys;
};

const getStatus = (statusNumber: number): Status => {
  switch (statusNumber) {
    case 1:
      return { Icon: Level1, sema: "sema10" };
    case 2:
      return { Icon: Level2Average, sema: "sema11" };
    case 3:
      return { Icon: Level3Bad, sema: "sema12" };
    case 4:
      return { Icon: Level4, sema: "sema13" };
    case 5:
      return { Icon: Level5, sema: "sema14" };
    default:
      return { Icon: null, sema: "sema1" };
  }
};

export { configuration, actions, getStatus };
