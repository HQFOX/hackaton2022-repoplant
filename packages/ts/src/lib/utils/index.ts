export const getRandom = (max: number, min = 1) =>
  Math.floor(Math.random() * (max - min + 1)) + min;
