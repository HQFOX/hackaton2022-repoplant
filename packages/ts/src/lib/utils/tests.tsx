import React from "react";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import { MemoryRouter } from "react-router-dom";
import configureMockStore from "redux-mock-store";
import { HvProvider } from "@hv/uikit-react-core";
import { TestAppState } from "typings/state";

export const createMockStore = (state: TestAppState): any => {
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const store = mockStore(state);
  return store;
};

export const withStoreProvider = <P extends {}>(
  Component: React.ComponentType<P>,
  state: TestAppState
): React.FC<P> => props => (
  <Provider store={createMockStore(state)}>
    <HvProvider>
      <MemoryRouter initialEntries={[{ pathname: "/", key: "testKey" }]}>
        <Component {...props} />
      </MemoryRouter>
    </HvProvider>
  </Provider>
);

export const withHvProvider = <P extends {}>(
  Component: React.ComponentType<P>
): React.FC<P> => props => (
  <HvProvider>
    <MemoryRouter initialEntries={[{ pathname: "/", key: "testKey" }]}>
      <Component {...props} />
    </MemoryRouter>
  </HvProvider>
);
