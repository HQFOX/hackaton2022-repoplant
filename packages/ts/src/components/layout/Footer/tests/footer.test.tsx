import React from "react";
import { mount } from "enzyme";
import { withHvProvider } from "lib/utils/tests";
import Footer from "../index";

describe("<Footer />", () => {
  let snapshot;

  beforeEach(() => {
    const WithHvProvider = withHvProvider(Footer);
    snapshot = mount(<WithHvProvider />);
  });

  test("it matches the snapshot", () => {
    expect(snapshot).toMatchSnapshot();
  });
});
