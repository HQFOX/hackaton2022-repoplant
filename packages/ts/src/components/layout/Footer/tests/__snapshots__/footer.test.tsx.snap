// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> it matches the snapshot 1`] = `
<div
  className="HvFooter-root"
>
  <p
    className="HvFooter-labelLeft HvTypography-root HvTypography-normalText"
  >
    Hitachi Vantara
  </p>
  <p
    className="HvFooter-labelRight HvTypography-root HvTypography-normalText"
  >
    Â© Hitachi Vantara Corporation 2020. All Rights Reserved.
  </p>
</div>
`;
